### PR TITLE
HESTData: update `to_spatial_data()```

### PR DESCRIPTION
**Pull request by konst-int-i**
## This PR

Makes several changes to the `HESTData.to_spatial_data()` behaviour: 
* Converts images to different resolutions(`fullres`, `hires_downsampled` and `lowres_downsampled`). 
* Saves full resolution version as `datatree` to prevent buffer error raised in #61

## Test instructions

```
from hest import load_hest
hest_data = load_hest('../hest_data', id_list=['TENX68'])
st = hest_data[0]
st.to_spatial_data(fullres=True)
```

Alternatively, run the first cell in `tutorials/2-Interacting-with-HEST-1k.ipynb`. 

## Expected behaviour

Old `SpatialData` conversion for sample `TENX68`

```
SpatialData object
├── Images
│     └── 'he': DataArray[cyx] (3, 38232, 20690)
├── Shapes
│     ├── 'cellvit': GeoDataFrame shape: (77165, 3) (2D shapes)
│     └── 'tissue_contours': GeoDataFrame shape: (2, 2) (2D shapes)
└── Tables
      └── 'anndata': AnnData (1657, 18085)
with coordinate systems:
    ▸ 'global', with elements:
        he (Images), cellvit (Shapes), tissue_contours (Shapes)
```
⬇️ ⬇️ ⬇️ 

New `SpatialData` conversion: 

```
SpatialData object
├── Images
│     ├── 'ST_downscaled_hires_image': SpatialImage[cyx] (3, 4779, 2586)
│     ├── 'ST_downscaled_lowres_image': SpatialImage[cyx] (3, 1000, 541)
│     └── 'ST_fullres_image': DataTree[cyx] (3, 38232, 20690), (3, 19116, 10345)
├── Shapes
│     ├── 'cellvit': GeoDataFrame shape: (77165, 3) (2D shapes)
│     ├── 'locations': GeoDataFrame shape: (1657, 2) (2D shapes)
│     └── 'tissue_contours': GeoDataFrame shape: (2, 2) (2D shapes)
└── Tables
      └── 'table': AnnData (1657, 18085)
with coordinate systems:
    ▸ 'ST_downscaled_hires', with elements:
        ST_downscaled_hires_image (Images), cellvit (Shapes), locations (Shapes), tissue_contours (Shapes)
    ▸ 'ST_downscaled_lowres', with elements:
        ST_downscaled_lowres_image (Images), cellvit (Shapes), locations (Shapes), tissue_contours (Shapes)
    ▸ 'ST_fullres', with elements:
        ST_fullres_image (Images), cellvit (Shapes), locations (Shapes), tissue_contours (Shapes)
 ```

### CODE REVIEW

File: src/hest/HESTData.py [hunk 1]
```diff
@@ -12,6 +12,9 @@
 from hestcore.wsi import (WSI, CucimWarningSingleton, NumpyWSI,
                           contours_to_img, wsi_factory)
 from loguru import logger
+from spatialdata import SpatialData
```
Comment (pauldoucet) [inline src/hest/HESTData.py:15]: We need to move this import line inside the to_spatial_data() method, the SpatialData is huge

Comment (konst-int-i) [reply src/hest/HESTData.py]: Moved in [98d0acf](https://github.com/mahmoodlab/HEST/pull/63/commits/98d0acfb87a8ecec1f5dca5fdee00032735900a7) - I had it outside to enable the type hint in the function, but appreciate that it comes with a lot of overhead

File: src/hest/HESTData.py [hunk 2]
```diff
@@ -460,60 +463,211 @@ def save_tissue_vis(self, save_dir: str, name: str) -> None:
...
+                tissue_lowres_scalef = None
+                hires = None
+                lowres = None
+                if SCALEFACTORS in keys:
+                    scalefactors = self.adata.uns[SPATIAL][dataset_id][SCALEFACTORS]
+                    if TISSUE_HIRES_SCALEF in scalefactors:
+                        tissue_hires_scalef = scalefactors[TISSUE_HIRES_SCALEF]
+                    else: 
+                        pixel_size=self.meta['pixel_size_um_estimated']
+                        ds_factor = 4/pixel_size # proxy for visium hires scale factor
```
Comment (pauldoucet) [inline src/hest/HESTData.py:557]: Is this part going to work for non-visium ST?

Comment (konst-int-i) [reply src/hest/HESTData.py:553]: Yes - the `hires` image here is an approximation for ~4 MPPs which is standard with Visium. I have tested it with non-visium slides and the same logic works and accurately downsamples the image, although the `hires` terminology is not normally given for these slides to my knowledge.

**Comment by konst-int-i** [comment]
Addressed the additional requests from @guillaumejaume in [82f0779](https://github.com/mahmoodlab/HEST/pull/63/commits/82f0779f188a257becf75de681c2be5b9ad09ba1)

**Review by guillaumejaume — APPROVED** [review]

**Pull request merged**